javascript/javascript3/week1/Weather_app/weather-app.js

```diff
@@ -0,0 +1,108 @@
+// Using: Async Await
+
+//Select the city, click Todays weather to get weather data
+const getData = document.querySelector("#search");
+
+// Click Current Location Button, to get latitude and longitude
+const currentLocationBtn = document.querySelector("#current-location")
+console.log(currentLocationBtn)
+
+// All display variables
+const displayLocation = document.querySelector(".location");
+const displayWeatherIcon = document.querySelector(".weather-image");
+const displayWeatherDescription = document.querySelector(".weather-description");
+const displayTemperature = document.querySelector(".temperature");
+const displayMinimumTemperature = document.querySelector(".minimum-temperature");
+const displayMaximumTemperature = document.querySelector(".maximum-temperature")
+const displayFeelLike = document.querySelector(".feels-like");
+const displayWindSpeed = document.querySelector(".wind");
+const displayCloudy = document.querySelector(".cloudy");
+const displaySunriseTime = document.querySelector(".sunrise");
+const displaySunsetTime = document.querySelector(".sunset");
+const displayLatitude = document.querySelector(".location-lat");
+const sunriseSunetAnimation = document.querySelector(".display-sunrise-sunset");
+const displayTodaysWeather = document.querySelector(".display-weather");
+
+const todaysWeatherData = async () => {
+    // Get the input value
+    let location = document.querySelector("#city").value;
+
+    // If location is blank, then default location = "Copenhagen"
+    if (location !== ""){
```

Comment:
`if (!str?.length)` also works

Reply:
Please explain this one... if (!str?.length)

Reply:
The ?. accesses the property length if it exists, so this statement runs if location is null or if it has a property length that is 0.